### ISSUE TITLE
[FEATURE REQUEST] - Add additional info columns to main tables

### ISSUE THREAD

**Issue by rgobo** [ebkr/r2modmanPlus#624]
**What problem does the feature request solve?**
Detecting potentially obsolete installed mods, helping with organization of mods, providing more information at a glance when either managing installed mods or looking for new ones to install.

**Example mock-up (optional)**
Current 4 columns are: Icon, Title, Enable/disable mod switch, Expand arrow.
Proposed additional (hideable in settings) columns: 
- Author (separate from title so it can be sorted)
- Date last updated
- Date created
- Version
- Downloads
- Rating

**Additional information (optional)**
If users could customize which columns to see, and/or re-order them, that'd be neat as well (for example, ability to put the toggle column as the first one rather than the second-to-last one).

**Comment by ebkr** [comment]
It's quite difficult to know when a mod is obsolete.

Most game updates probably won't break a mod unless it touches an area that has changed. Let's take the NoBossNoWait mod for RoR2, that hasn't been updated in over a year and still works as intended.

Downloads/Rating/Date created/Date last updated aren't particularly valid columns because they may not exist for all mods. Locally imported mods for example unfortunately wouldn't have this information collectable.

I could add an alphabetical author name sort if desirable though.

**Comment by rgobo** [comment]
I know it's not a perfect metric but precisely because there is no sure way to know, having more info to make an educated guess helps, or at least narrows down possible issues. With the recent valheim update for example, which broke many things, it would be nice to be able to sort by date and see the cut-off of mods updated before or after the new version.

I didn't think about locally installed mods, but I guess that's always going to be an issue if there's no metadata file associated with them. Maybe just leave the columns empty and provide info for thunderstore mods that have the data instead. It's kinda weird being able to sort by downloads or rating, but not actually see what the values are right there. The last updated date is shown in the expand as well already, as well as categories, but there's no easy way to see the actual value without clicking on every item individually. If the data is already there, why not show it as well, when possible?

**Comment by ebkr** [comment]
Ah you might be slightly confusing the online and installed tabs.

There’s no last updated / categories on the installed screen, those are all in the online tab. Once it’s installed that metadata is only accessible when comparing the author/ModName stuff to what’s in the online list (which would break if not online and no local api cache)

**Comment by rgobo** [comment]
I admit I haven't looked at the code yet and I don't know what data is stored or not, I assumed it is downloading/caching the basic metadata when a mod is installed. For example, it can show mod version from the thunderstore, but you're saying for other metadata it would have to load it again / match it to online repo? 

![image](https://user-images.githubusercontent.com/9076751/134906834-faef42f9-351b-4a42-a68c-eb70cb836340.png)

Meanwhile the online tab has a different set of information:
![image](https://user-images.githubusercontent.com/9076751/134906906-337f3065-0a9d-42db-a4fe-1c000a5d17fc.png)

If you say it is not so simple to interchangeably share data between these two tabs I believe you, but it might still be nice to be able to toggle at least the available stuff then. 
In any case, just to clarify the request: my proposal is to show the stuff from the expand window, technical issues notwithstanding, in the collapsed grid for easier searching. Probably as an optional toggle in settings if the desired default design is to keep it as simple and minimalistic as possible. 

![image](https://user-images.githubusercontent.com/9076751/134907258-0e517d6a-c486-482a-a86f-e34de2be45fd.png)

**Comment by ebkr** [comment]
Yeah I mean it's possible to share the two although it's a bit weird regarding locally installed mods. Having the data not always being available is weird and is potentially confusing. Even if that data can be found, it could be outdated and/or irrelevant (EG: category may change to include another kind, such as an enemy mod that now includes a playable character).

Installed and online listings actually have different sets of data, so basically there's potential to "bridge" between the two, but it's not guaranteed to have all the data. In terms of understanding what data is generally available in the installed section, take a look inside a manifest.json file on TS.

Assuming we get past that, the next problem would be interface. I'm not really sure if it can afford to use that much space given that "_package name_ **_by_** _author name_" can be relatively long. It also wouldn't scale too well on smaller windows.
Additionally I'm not too sure how column visiblity would work without being confusing.

FWIW they also aren't currently separated into columns, it's pretty much just hacked together. I could probably separate them into individual columns without too much trouble though.
